django/forms/fields.py

```diff
@@ -988,11 +988,12 @@ def __init__(self, fields, *, require_all_fields=True, **kwargs):
                                         self.error_messages['incomplete'])
             if self.disabled:
                 f.disabled = True
-            if self.require_all_fields:
-                # Set 'required' to False on the individual fields, because the
-                # required validation will be handled by MultiValueField, not
-                # by those individual fields.
-                f.required = False
+            if (self.required and (self.require_all_fields or f.required)) or (
+                not self.required and not self.require_all_fields and f.required
+            ):
+                f.required = f.widget.is_required = True
+            else:
+                f.required = f.widget.is_required = False
```

Comment:
The conditional here needs a comment at least. (Did we change the rules here?)

Reply:
I'm trying to set `is_required` on the widget, so we can us this later on. My realisation tonight is here I'm changing `is_required` on the wrong widget. Here I'm setting it on `field.widget`. What I need to do is set it on `widget.widets.widget`. Or more simply, it needs to set on the widgets of the MultiWidget rather than the widgets of the fields of the MultiField. 

I'm hoping this will simplify all of this a great deal as we can 'just' rely on `get_context` to do the right thing if we set it up right here. 🤞 

Leave it with me and I'll revert after some more thought.